meilisearch-core/src/bucket_sort.rs

```diff
@@ -328,13 +330,20 @@ where
                     };
 
                     if filter_accepted {
+                        let contains_key = key_cache.contains_key(&document.id);
                         let entry = key_cache.entry(document.id);
                         let key = entry.or_insert_with(|| (distinct)(document.id).map(Rc::new));
 
-                        match key.clone() {
+                        let distinct_accepted = match key.clone() {
                             Some(key) => buf_distinct.register(key),
                             None => buf_distinct.register_without_key(),
                         };
+
+                        if !distinct_accepted && !contains_key {
```

Comment:
why do you need to check contain_key?

Reply:
Ah, now I see this is a mistake. I noticed that when I did the `filter_accepted` check, I needed the `contains_key` since the document IDs were repeated in the groups. If I didn't do the `contains_key` for filter, then I got overall `nbHits` as 0 since `filtered_count` was high. I think I don't need this for `distinct_accepted` like you pointed out. I will remove it.